ethereum/stratum/src/main/java/org/hyperledger/besu/ethereum/stratum/Stratum1EthProxyProtocol.java

```diff
@@ -49,9 +51,14 @@
   private final MiningCoordinator miningCoordinator;
   private EthHashSolverInputs currentInput;
   private Function<EthHashSolution, Boolean> submitCallback;
+  private final EpochCalculator epochCalculator;
 
   public Stratum1EthProxyProtocol(final MiningCoordinator miningCoordinator) {
+    if (!(miningCoordinator instanceof EthHashMiningCoordinator)) {
+      throw new IllegalArgumentException();
```

Comment:
We need a message in this throw, something like `"Stratum1 Proxies require an EthHashMiningCoordinator not " + (miningCoordinator == null) ? "null" : miningCoordinator.getClass().getName()`